### PR TITLE
add tutorial - setting up LXD on 16.04

### PR DESCRIPTION
**Pull request by albertodonato**
Add a tutorial on setting up LXD on Ubuntu 16.04 Xenial Xerus.

### CODE REVIEW

File: tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+
+If you're running an Ubuntu 16.04 server installation, LXD is already
+preinstalled on your system.
+
+On Desktop installs, it needs to be installed manually. Open a terminal and run
+the following commands to install LXD:
+
+```bash
+$ sudo apt update
```
Comment (evilnick) [inline tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md]: We don't include the prompt, and separate commands from output - see https://github.com/canonical-websites/tutorials.ubuntu.com/blob/master/examples/guidelines-snap-tutorials.md#command-lines

Sorry if you have copied this from some existing tutorials - a few were bulk added and we need to retroactively make them fit the style

File: tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+To do so, run
+
+```bash
+$ sudo adduser <USER> lxd
+```
+
+where `<USER>` is your username.
+
+The new group will be effencive at the next login session, to apply it to the
```
Comment (evilnick) [inline tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md]: effective?

File: tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+which asks a series of questions on how to configure the daemon.
+
+We can just go with the default answer (by pressing Enter) for most of
+them. You can pick the size of the loop device for the ZFS pool based on your
+available disk space.
+
+![screenshot](images/lxd-init-1.png)
+
+
+As part of the LXD init setup, networking can be also set up, so that spawned containers can be accessed from the host.
```
Comment (evilnick) [inline tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md]: Try to keep text wrapped to 79 characters

File: tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+Once the container is no longer needed, it can be stopped and destroyed:
+
+```bash
+$ lxc stop stirring-beagle
+$ lxc delete stirring-beagle
+```
+
+
+## Finding help
```
Comment (evilnick) [inline tutorials/setting-up-lxd-1604/setting-up-lxd-1604-tutorial.md]: This is the final page and should have some sort of recap of what has been covered, as well as some links to other things of interest.

The 'Finding Help' idea is a good one, but on it's own seems a little negative.  Have a look at https://tutorials.ubuntu.com/tutorial/snap-python-app#4

**Comment by albertodonato** [comment]
@evilnick all comments should be addressed now. thanks!

**Comment by evilnick** [comment]
That's awesome, thanks so much!

:cake:

**Comment by evilnick** [comment]
@caldav do you want to take a quick look at this in case I missed something?

**Comment by caldav** [comment]
Hey @albertodonato, the tutorial is a great addition! 

Have you tried running it locally?

`./run serve tutorials/setting-up-lxd-1604`

When doing so, I see a couple things that still needs a fix:

* Please avoid wrapping non-code lines at 80 chars, it makes the content look very different from other tutorials and creates unexpected line breaks on mobile
* The logo on the first page is full-width, which makes it look oversized and pixelated. Also, it seems that the MD parser doesn't support resizing images and always goes for full-width, until this is fixed, would you mind using this version instead? 

containers-large

(same logo, with extra padding on the right)

Thanks

**Comment by caldav** [comment]
Another question: any reason why the tutorial is not using the LXD snap?

**Comment by albertodonato** [comment]
@caldav thanks for the review.

I didn't notice that the MD parser actually preserves line breaks. I thought it would consider text without an empty line as a single paragraph, as github does. It should be fixed now

I've replaced the image.

The tutorial doesn't use the snap since it's not yet production-ready.
An alternative to get a more recent version of LXD is to get it from xenial-backports, but it didn't really matter for the purpose of this tutorial, so I based on the version in main.

**Comment by caldav** [comment]
@albertodonato thanks, this md parser is a bit special in that regard, and I would have assumed the same thing. I'll follow up on a bug report for this, as it would make sense.

Thanks for the explanation, makes sense!

**Pull request merged**